external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaTupleListener.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.storm.kafka.spout;
+
+import java.io.Serializable;
+import java.util.List;
+
+
+/**
+ * The KafkaTupleListener handles state changes of a kafka tuple inside a KafkaSpout.
+ */
+public interface KafkaTupleListener extends Serializable {
```

Comment:
I meant being able to emit the tuples that have exceeded their retry limit (those that hit onMaxRetryReached) to a stream. It's probably not a great idea, since we'd need to consider processing guarantees for those too, so never mind :)